src/ApacheOrcDotNet.OptimizedReaderTest.App/TradeDataSourceApp.cs

```diff
@@ -40,17 +46,24 @@ public void Run()
                 var times = new decimal?[approxRowCount];
                 var prices = new decimal?[approxRowCount];
                 var sizes = new long?[approxRowCount];
+                var numRows = 0;
 
-                int numRows = timeRangeReader.ReadBatch(times, prices, sizes);
+                while (true)
+                {
+                    var rowsRead = timeRangeReader.ReadBatch(numRows, times, prices, sizes);
```

Comment:
Rather than passing `writeStartIndex`, I think you should slice the Spans on this side.